### PR TITLE
Tests for MerkleProof.sol #2

### LINKED ISSUE #2: Tests for MerkleProof.sol
Write tests in `packages\hardhat\test\MerkleProof.test.ts` for the modified `MerkleProof.sol` library we use.

As the implementation is still very similar to OZ's, we should be able to re-use a lot of their tests: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v3.4.0/test/cryptography/MerkleProof.test.js

### PR DESCRIPTION
**Pull request by IoanDraghici**
### What does it do?
Tests for the MerkleProof.sol library.

### Any helpful background information?
None

### Any new dependencies? Why were they added?
None

### Does it close any issues?

Closes #2

### CODE REVIEW

**Comment by janndriessen** [comment]
I'll take a closer look tomorrow or the day after. @IoanDraghici please run prettier on your files in the meanwhile - as the building process found some issues.

File: packages/hardhat/test/MerkleProof.test.ts [hunk 1]
```diff
@@ -1,3 +1,81 @@
...
+      const leaf: Buffer = generateLeaf(formattedAddress, numTokens);
+      const proof: string[] = this.merkleTree.getHexProof(leaf);
+
+      const result = await this.merkleProof.verify(proof, this.merkleRoot, leaf);
+
+      expect(result[0]).to.be.true;
+      expect(result[1].toNumber()).to.equal(0);
+    });
+
+    it('returns true and index 1 for a valid Merkle proof', async function () {
```
Comment (janndriessen) [inline packages/hardhat/test/MerkleProof.test.ts:45]: `returns true and index 1` should say `returns true and index 2`. The expect expression is also actually testing for `2`.

**Review by janndriessen — CHANGES_REQUESTED** [review]

**Comment by janndriessen** [comment]
When the comment is fixed, @embiem can review as well or merge right away.

File: packages/hardhat/test/MerkleProof.test.ts (hunk 1 above)
Comment (IoanDraghici) [reply packages/hardhat/test/MerkleProof.test.ts]: fixed

**Review by janndriessen — APPROVED** [review]
🚀

**Pull request merged**